### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by rubydevi**
### Description:
In this step, I have implemented [Redux Toolkit](https://redux-toolkit.js.org/) inside of the application. 
**I have:**
- Setup a Redux store
- Created a slice to save the state and reducers

**Changed Made:**

- Added Redux Toolkit
- Created a directory that contains all the Redux logic (`/src/redux`)
- Configured a Redux store (`/src/redux/store.js`)
- Defined a `slice` of state for `books` that:
  -   Includes an array of books (initial state: empty array)
  -   Includes a reducer that adds a book
  -   Includes a reducer that removes a book
- Defined a `slice` of state for `categories` that:
  - Includes an array of categories (initial state: empty array)
  - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

Structured the application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for all Redux files like this:
```
/src
    |---/components
    |---/redux
        |--- /books
                | booksSlice.js
        |--- /categories
                | categoriesSlice.js
        | store.js
    | index.js
```

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => {
+      state.categories = ['Under construction'];
+    },
+  },
+});
```
Comment (AbelG101) [inline src/redux/categories/categoriesSlice.js]: - [ ] Great work so far! However, according to the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md), the reducer inside the categories slice needs to check the status before updating the state. It appears that this step has been missed in your code, so please ensure that you check the payload from the action before making any updates to the state, Here is an example of how to do this:
```jsx
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories

```
- [ ] As you might have guessed this also means you'll have to add a second parameter called `action` to the  `checkStatus` reducer.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @rubydevi 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Redux & react-redux are properly setup ✔
- Proper use of reducers ✔
- Correct use of Git Flow ✔
- Top-notch readme documentation ✔
- Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (rubydevi) [reply src/redux/categories/categoriesSlice.js]: Thank you so much for your valuable feedback @AbelG101 , I really appreciate it.  I will make the changes according to that.

**Review by danmainah — APPROVED** [review]
Hi @rubydevi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Implemented the change requested to ensure the categories reducer is properly configured.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**